test/http_downloads_test.rb

```diff
@@ -7,8 +7,14 @@ def command(src, dst)
 end
 
 class HttpDownloadsTest < Test::Unit::TestCase
+  def test_should_return_an_instance
+    assert_instance_of Proxy::HttpDownload, Proxy::HttpDownloads.start_download('source', 'destination')
+  ensure
+    File.delete('destination')
```

Comment:
Unless you join the resulting HttpDownload, could this be executing still in the background of the test?  Perhaps better to stub it like tftp_api_test?